packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -103,7 +104,10 @@ class TreeItem extends UIComponent<ReactProps<TreeItemProps>> {
 
   handleTitleOverrides = (predefinedProps: TreeTitleProps) => ({
     onClick: (e, titleProps) => {
-      e.preventDefault()
+      // Only prevent default if the key is spacebar
+      if (keyboardKey.getCode(e) === keyboardKey.Spacebar) {
+        e.preventDefault()
+      }
```

Comment:
we try to omit using keys explicitly in Stardust components. 
Instead, put `e.preventDefault` in actionHandlers. See comment below